pkgs/applications/science/logic/z3/default.nix

```diff
@@ -43,6 +43,13 @@ let common = { version, sha256, patches ? [ ], tag ? "z3" }:
     postPatch = optionalString ocamlBindings ''
       export OCAMLFIND_DESTDIR=$ocaml/lib/ocaml/${ocaml.version}/site-lib
       mkdir -p $OCAMLFIND_DESTDIR/stublibs
+    '' + optionalString ((versionAtLeast python.version "3.12") && (versionOlder version "4.8.14")) ''
+      # See https://github.com/Z3Prover/z3/pull/5729. This is a specialization of this patch for 4.8.5.
+      for file in scripts/mk_util.py src/api/python/CMakeLists.txt; do
+        substituteInPlace "$file" \
+          --replace-fail 'distutils.sysconfig.get_python_lib()' "sysconfig.get_path('purelib')" \
+          --replace-fail 'distutils.sysconfig' 'sysconfig'
+      done
```

Comment:
Another option I see is to set CMAKE_INSTALL_PYTHON_PKG_DIR to the sitePackages directory (see: https://github.com/Z3Prover/z3/pull/5729/files#diff-03fa5bcf870844dbf2dc0a66027112e4f0ff0dd5c5cd5a7f946b1567b240be37L98-L121), but we'd still have to do this substituteInPlace for mk_util.py on pre-4.8.14 Z3 and we'd rebuild every z3 version instead of just pre-4.8.14.